src/object/object_representation.c

```diff
@@ -8402,3 +8479,54 @@ htond (double from)
 #endif /* ! OR_HAVE_HTOND */
 
 #endif /* OR_BYTE_ORDER == OR_LITTLE_ENDIAN */
+
+/* or_get_varchar_comp_lengths()  :   Function to get the compressed length
+ *				      and the uncompressed length of a compressed string.
+ *
+ * return			  :   NO_ERROR or error_code.
+ * buf(in)			  :   The buffer where the string is stored.
+ * compressed_size(out)		  :   The compressed size of the string. Set to -1 if the string was not compressed.
+ * decompressed_size(out)	  :   The uncompressed size of the string.
+ */
+
+int
+or_get_varchar_compression_lengths (OR_BUF * buf, int *compressed_size, int *decompressed_size)
+{
+  int compressed_length = 0, decompressed_length = 0, rc = NO_ERROR;
+  int size_prefix = 0;
+
+  /* Make sure the string is compressed */
+  size_prefix = or_get_byte (buf, &rc);
+  if (rc != NO_ERROR)
+    {
+      assert (size_prefix == 0);
+      return rc;
+    }
+
+  if (size_prefix == 0xFF)
+    {
+      /* String was compressed */
+      /* Get the compressed size */
+      rc = or_get_data (buf, (char *) &compressed_length, OR_INT_SIZE);
+      if (rc != NO_ERROR)
```

Comment:
@lucianlazar2211 

Please find my comments on `or_put_varchar_internal`. 
When you change it, endian conversions are also required here.